test/perf_counters_gtest.cc

```diff
@@ -97,98 +98,136 @@ TEST(PerfCountersTest, Read2Counters) {
 }
 
 TEST(PerfCountersTest, ReopenExistingCounters) {
-  // The test works (i.e. causes read to fail) for the assumptions
-  // about hardware capabilities (i.e. small number (3-4) hardware
-  // counters) at this date.
+  // This test works as you can have multiple repeated counters
+  // in different performance counter groups
   if (!PerfCounters::kSupported) {
     GTEST_SKIP() << "Test skipped because libpfm is not supported.\n";
   }
   EXPECT_TRUE(PerfCounters::Initialize());
-  std::vector<PerfCounters> counters;
-  counters.reserve(6);
-  for (int i = 0; i < 6; i++)
-    counters.push_back(PerfCounters::Create({kGenericPerfEvent1}));
+  std::vector<std::string> kMetrics({kGenericPerfEvent1});
+  std::vector<PerfCounters> counters(6);
+  for (auto& counter : counters) {
+    counter = PerfCounters::Create(kMetrics);
+  }
   PerfCounterValues values(1);
   EXPECT_TRUE(counters[0].Snapshot(&values));
-  EXPECT_FALSE(counters[4].Snapshot(&values));
-  EXPECT_FALSE(counters[5].Snapshot(&values));
+  EXPECT_TRUE(counters[4].Snapshot(&values));
```

Comment:
i'm not sure what this is testing now if these are just all valid

Reply:
Yes, I agree. This is one of the tests that were previously failing. On older Intel architectures there was a limit of 3 PMCs while in newer ones there are 6 PMCs so these tests were not returning false as expected. And new hardware PMCs are being introduced with every new launch [see the AMD Zen4](https://www.phoronix.com/news/LInux-6.2-AMD-Zen-4-Events) so there is really not much sense in keeping these tests as they are, in fact.

Reply:
True. This entire PR is hanging now on a green light on the core changes. There is really a substantial amount of clean up to be done if true, including comments.

Reply:
I removed the tests for `N>3` since it would be impossible to make any reasonable assumption regarding the current hardware without explicitly querying libpfm.